services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/PinnedFlowSpec.groovy

```diff
@@ -71,12 +80,20 @@ class PinnedFlowSpec extends BaseSpecification {
         cookiesMap.sort() == cookiesMapAfterReroute.sort()
         metersMap.sort() == metersMapAfterReroute.sort()
 
-        when: "The broken ISL is restored"
-        northbound.portUp(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+        when: "The broken ISLs are restored one by one"
+        islsToBreak[0..-2].each { islToBreak ->
+            northbound.portUp(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
+            Wrappers.wait(WAIT_OFFSET + discoveryInterval) {
+                assert islUtils.getIslInfo(islToBreak).get().state == IslChangeType.DISCOVERED
```

Comment:
There is a reroute delay after ISL is discovered, so these check may not catch the problem if it appears

Reply:
fixed, added: `TimeUnit.SECONDS.sleep(rerouteDelay - 1)`